### DISCUSSION TITLE
Route::apiResource for update

### DISCUSSION THREAD

**Discussion by muazkhairi92** [laravel/framework#50276] [Q&A] [answered]
Currently, if we want to use update method without an id, we need to separate it from Route::apiResource.
` Route::apiResource('example', ExampleController::class)->only(['store','index']);
`
`Route::put('example', [ExampleController::class,'update']);
`

Is there any way that we can use ` Route::apiResource('example', ExampleController::class)` without an id for update?

**Comment by ghost** [comment]
You want mass update? or composite pk update?

**Reply by muazkhairi92** [reply]
I want to update based on what belongs to a user
```
   public function update(Request $request)
   {
       $example = Auth::user()->example;
       $example->update($request->all());
       return $example;
   }
```

**Reply by ghost** [reply]
if the user is authenticated the FE should know its pk. Don't forget to validate your request.

**Reply by muazkhairi92** [reply]
Sorry if the question is unclear.
My function method works perfectly fine without sending any id(pk).
This endpoint works: `Route::put('example', [ExampleController::class,'update']);` .
However, the update function method did not work in  `Route::apiResource('example', ExampleController::class)`
Is there a way so that I can use `Route::apiResource` without requiring any id for the update method?

**Reply by ghost** [reply, marked as answer]
There is if you code it like you did. REST is clear about updates. You need an identifier.

**Reply by ghost** [reply]
Or send a dummy identifier and overwrite the update method to ignore it.

**Reply by muazkhairi92** [reply]
I never know that REST always need an identifier for update method. Thank you for informing

**Reply by muazkhairi92** [reply]
Then, maybe also in laravel `Illuminate\Routing\ResourceRegistrar` this can be adjusted.

because in `addResourceUpdate` it requires an identifier but in `addSingletonUpdate` it does not.
```
    /**
     * Add the update method for a resourceful route.
     *
     * @param  string  $name
     * @param  string  $base
     * @param  string  $controller
     * @param  array  $options
     * @return \Illuminate\Routing\Route
     */
    protected function addResourceUpdate($name, $base, $controller, $options)
    {
        $name = $this->getShallowName($name, $options);

        $uri = $this->getResourceUri($name).'/{'.$base.'}';

        $action = $this->getResourceAction($name, $controller, 'update', $options);

        return $this->router->match(['PUT', 'PATCH'], $uri, $action);
    }
```


```
    /**
     * Add the update method for a singleton route.
     *
     * @param  string  $name
     * @param  string  $base
     * @param  string  $controller
     * @param  array  $options
     * @return \Illuminate\Routing\Route
     */
    protected function addSingletonUpdate($name, $controller, $options)
    {
        $name = $this->getShallowName($name, $options);

        $uri = $this->getResourceUri($name);

        $action = $this->getResourceAction($name, $controller, 'update', $options);

        return $this->router->match(['PUT', 'PATCH'], $uri, $action);
    }
```

**Reply by ghost** [reply]
I find the laravel apiResource very basic. It does not have filtering capabilities. I proposed something more advanced like https://github.com/laravel/framework/discussions/49577  in this question https://github.com/laravel/framework/discussions/50243 but it gets 0 interest.

**Reply by muazkhairi92** [reply]
Great idea there, hope it can be implemented
